File: code/backend/src/common/enums.ts
export enum Role {
  OWNER = 'OWNER',
  STAFF = 'STAFF',
}

export enum OrderStatus {
  NEW = 'NEW',
  PREPARING = 'PREPARING',
  SERVED = 'SERVED',
  PAID = 'PAID',
  CANCEL = 'CANCEL',
}

export enum PaymentMethod {
  CASH = 'CASH',
  TRANSFER = 'TRANSFER',
}

export enum TableStatus {
  AVAILABLE = 'AVAILABLE',
  CLOSED = 'CLOSED',
  OCCUPIED = 'OCCUPIED',
}

export enum PaymentMethod {
  VNPAY = 'VNPAY',
}

export enum PaymentStatus {
  PENDING = 'PENDING',
  SUCCESS = 'SUCCESS',
  FAILED = 'FAILED',
}

File: code/frontend/mobile/lib/data/models/enums.dart
enum OrderStatus {
  newOrder('NEW'),
  preparing('PREPARING'),
  served('SERVED'),
  paid('PAID'),
  cancel('CANCEL');

  final String value;
  const OrderStatus(this.value);

  factory OrderStatus.fromString(String val) {
    final normalized = val.trim().toUpperCase();
    return OrderStatus.values.firstWhere(
      (e) => e.value == normalized,
      orElse: () => OrderStatus.newOrder,
    );
  }
}

enum PaymentMethod {
  cash('CASH'),
  transfer('TRANSFER');

  final String value;
  const PaymentMethod(this.value);

  factory PaymentMethod.fromString(String val) {
    return PaymentMethod.values.firstWhere(
      (e) => e.value == val,
      orElse: () => PaymentMethod.cash,
    );
  }
}

enum UserRole {
  owner('OWNER'),
  staff('STAFF');

  final String value;
  const UserRole(this.value);

  factory UserRole.fromString(String val) {
    return UserRole.values.firstWhere(
      (e) => e.value == val,
      orElse: () => UserRole.staff,
    );
  }
}
